app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala

```diff
@@ -111,11 +113,14 @@ class DesConnector @Inject() (httpClient: HttpClient, metrics: Metrics, agentCac
     }
   }
 
-  private def desHeaders(authToken: String, env: String) = Seq(
-    HeaderNames.authorisation -> s"Bearer $authToken",
-    Environment               -> env,
-    CorrelationId             -> UUID.randomUUID().toString
-  )
+  def desHeaders(authToken: String, env: String)(implicit hc: HeaderCarrier): Seq[(String, String)] =
+    Seq(
+      HeaderNames.authorisation -> s"Bearer $authToken",
+      Environment               -> env,
+      CorrelationId             -> UUID.randomUUID().toString
+    ) ++
+      (if (hc.sessionId.map(_.value).nonEmpty) Seq(SessionId -> hc.sessionId.map(_.value).get) else Seq.empty) ++
```

Comment:
Could you please use map or fold instead ?? hc.sessionId.fold(Seq.empty[(String,String)])(x=>Seq(SessionId -> x.value)) or with map would be hc.sessionId.map(x=> Seq(SessionId -> x.value)).getOrElse(Seq.empty[(String,String)])